starlite/datastructures/cookie.py

```diff
@@ -33,23 +34,58 @@ class Cookie(BaseModel):
     documentation_only: bool = False
     """Defines the Cookie instance as for OpenAPI documentation purpose only"""
 
+    @cached_property
```

Comment:
Since `Cookie` is mutable, `dict` is cached as well, and `__hash__` does not include, this can lead to a stale cache being returned.

```python
c = Cookie(key="foo", value="bar", secure=False)
simple_cookie = c.simple_cookie
c.secure = True
simple_cookie is c.simple_cookie
```

Reply:
ok, ill switch this to property

Reply:
Then the problem still exists for `dict`. I think the right way would be to change `__hash__` or just use the built-in one from `dataclass`.

Reply:
i removed them all.